src/m365/spo/commands/page/page-section-add.ts

```diff
@@ -101,6 +166,40 @@ class SpoPageSectionAddCommand extends SpoCommand {
           }
         }
 
+        if (typeof args.options.iconAlignment !== 'undefined') {
+          if (SpoPageSectionAddCommand.IconAlignment.indexOf(args.options.iconAlignment) === -1) {
+            return 'The value of parameter iconAlignment must be Left|Right';
+          }
+        }
+
+        if (typeof args.options.fillMode !== 'undefined') {
+          if (SpoPageSectionAddCommand.FillMode.indexOf(args.options.fillMode) === -1) {
+            return 'The value of parameter fillMode must be ScaleToFill|ScaleToFit|Tile|OriginalSize';
+          }
+        }
+
+        if (args.options.zoneEmphasis !== 'Image' && (args.options.imageUrl || args.options.imageWidth ||
+          args.options.imageHeight || args.options.fillMode)) {
+          return 'Specify imageUrl, imageWidth, or imageHeight only when zoneEmphasis is set to Image';
+        }
+
+        if (args.options.zoneEmphasis !== 'Gradient' && (args.options.gradientText)) {
+          return 'Specify gradientText only when zoneEmphasis is set to Gradient';
+        }
+
+        if (args.options.overlayOpacity && (args.options.overlayOpacity < 0 || args.options.overlayOpacity > 100)) {
+          return 'The value of parameter overlayOpacity must be between 0 and 100';
+        }
+
+        if (args.options.overlayColor && !/^#[0-9a-f]{6}$/i.test(args.options.overlayColor)) {
+          return 'The value of parameter overlayColor must be a valid hex color';
+        }
+
+        if ((args.options.zoneEmphasis !== 'Image' && args.options.zoneEmphasis !== 'Gradient')
+          && (args.options.overlayColor || args.options.overlayOpacity || args.options.useLightText)) {
+          return 'Specify overlayColor or overlayOpacity only when zoneEmphasis is set to Image or Gradient';
+        }
```

Comment:
This can be simplified a bit.

```suggestion
        if (!['image', 'gradient'].includes(args.options.zoneEmphasis.toLowerCase()) && (args.options.overlayColor || args.options.overlayOpacity || args.options.useLightText)) {
          return 'Specify overlayColor or overlayOpacity only when zoneEmphasis is set to Image or Gradient';
        }
```

Reply:
Done